crates/sui-e2e-tests/tests/reconfiguration_tests.rs

```diff
@@ -609,9 +648,260 @@ async fn test_reconfig_with_committee_change_basic() {
                 .epoch_store_for_testing()
                 .committee()
                 .num_members(),
-            4
+            initial_num_validators
+        );
+    });
+}
+
+#[sim_test]
+async fn test_switch_to_new_protocol_version() {
+    let initial_num_validators = 10;
+    let new_validator = ValidatorGenesisConfigBuilder::new().build(&mut OsRng);
+
+    let address = (&new_validator.account_key_pair.public()).into();
+    let mut test_cluster = TestClusterBuilder::new()
+        .with_protocol_version(PRE_SIP_39_PROTOCOL_VERSION.into())
+        .with_epoch_duration_ms(20000)
+        .with_accounts(vec![
+            AccountConfig {
+                gas_amounts: vec![DEFAULT_GAS_AMOUNT],
+                address: None,
+            },
+            AccountConfig {
+                gas_amounts: vec![DEFAULT_GAS_AMOUNT],
+                address: Some(address),
+            },
+        ])
+        .with_num_validators(initial_num_validators)
+        .build()
+        .await;
+
+    // adds a validator candidate which cannot enter the commitee before SIP-39
+    let stake = (DEFAULT_GAS_AMOUNT * (initial_num_validators as u64)) / 10_000 * 20;
```

Comment:
Could you add a comment that explains what this equation do, and why it would fail to enter the committee?